dev-lua/lua-openssl/files/0001-fix-libressl-compat.patch

```diff
@@ -0,0 +1,433 @@
+From 44d5360d0caed1f4f364691f789fe825fcd17a3e Mon Sep 17 00:00:00 2001
+From: crito <crito@fnordpipe.org>
+Date: Mon, 1 Oct 2018 20:10:05 +0200
+Subject: [PATCH] fix libressl compat
+
+add conditions to build with libressl and add missing functions.
+backported from upstream repo.
+---
+ src/compat.c  | 11 ++++++++---
+ src/digest.c  |  2 +-
+ src/ec.c      |  2 +-
+ src/engine.c  | 10 +++++-----
+ src/lhash.c   |  7 ++++---
+ src/openssl.c | 12 ++++++++++--
+ src/ots.c     |  5 +++++
+ src/pkcs7.c   | 24 ++++++++++++------------
+ src/private.h |  7 ++++---
+ src/sm2.c     |  2 +-
+ src/srp.c     |  3 ++-
+ src/x509.c    |  2 +-
+ 12 files changed, 54 insertions(+), 33 deletions(-)
+
+diff --git a/src/compat.c b/src/compat.c
+index cc4cc21..cc45845 100644
+--- a/src/compat.c
++++ b/src/compat.c
+@@ -5,7 +5,7 @@
+ #include "openssl.h"
+ #include "private.h"
+ 
+-#if OPENSSL_VERSION_NUMBER < 0x10100000L
++#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
```

Comment:
For the record, LibreSSL-2.8 probably needs another version condition here.

Reply:
interesting point. In the upstream repo it uses this condition for the libressl-2.8 fix. I'll compile libressl-2.8 later and test it.
Probably it requires another version condition in before line 40.

Reply:
yepp. @mgorny you're right. both function declarations are available in libressl-2.7.0.

Reply:
@mgorny & @gyakovlev on the first view I saw a lot of mismatching pointer types. It looks like there were a lot of changes in the API. From my point of view I would mark a hard dependency on <libressl-2.7.0 and check the next version of lua-openssl or try to work upstream on that. what do you think? because what I saw that will end in a lot of custom conditions between openssl 1.0.x / 1.1.x and libressl 2.6.x / 2.8.x

Reply:
Works for me. I thought libressl-2.8 was supposed to be more compatible with openssl-1.1 but meh.